lib/Service/SynchronizationService.php

```diff
@@ -56,70 +63,144 @@ public function __construct(
 	 * Synchronizes a given synchronization (or a complete source).
 	 *
 	 * @param Synchronization $synchronization
-	 * @return array
+	 * @param bool|null       $isTest False by default, currently added for synchronziation-test endpoint
+     * 
 	 * @throws Exception
+     *
+	 * @return array
 	 */
-    public function synchronize(Synchronization $synchronization): array
+    public function synchronize(Synchronization $synchronization, ?bool $isTest = false): array
```

Comment:
Grote functie, zou deze makkelijk/snel nog wat ingekort kunnen worden?

Reply:
Ja maar dat was die al, en functie clean up is natuurlijk geen onderdeel van de fix ;)

Reply:
Oke, prima voor nu, maar moeten we dan nog wel een andere keer naar kijken